ab3d2_source/hires.s

```diff
@@ -9854,30 +9851,42 @@ dosomething:
 				tst.b	PLR1_StoodInTop
 				beq.s	.okinbot
 				move.w	ZoneT_UpperFloorNoise_w(a0),d0
+
 .okinbot:
+				; Issue #1 - Check we are on the floor before applying any floor damage.
+				; TODO - add water checks. Water in areas with floor damage could be acid pools.
+				move.l	PLR1s_tyoff,d1
+				sub.l	PLR1s_yoff,d1
```

Comment:
Not that it matters much, but just wondering why using cmp wasn't an option here

Reply:
It seemed like more instructions for a value that was going to get thrown away almost immediately. When I come to consider water depth, it may be good to rethink it since there'll be a couple of things to compare to.

Reply:
Wait, I think I misunderstood. You mean cmp.l of the the memory operand directly with d1 - I may have brainfarted and been thinking about moving the other value to a different register first.

Reply:
Try again...